cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/TransformExecutorFactory.java

```diff
@@ -89,4 +90,14 @@ protected Transformation getTransformation(String pluginType, String stageName)
     return plugin;
   }
 
+  protected static <IN, OUT> TrackedTransform<IN, OUT> getTrackedGroupStep(Transformation<IN, OUT> transform,
+                                                                              StageMetrics stageMetrics) {
+    return new TrackedTransform<>(transform, stageMetrics, TrackedTransform.RECORDS_IN, null);
+  }
+
+  protected static <IN, OUT> TrackedTransform<IN, OUT> getTrackedAggregateStep(Transformation<IN, OUT> transform,
+                                                                               StageMetrics stageMetrics) {
+    return new TrackedTransform<>(transform, stageMetrics, "aggregator.groups", TrackedTransform.RECORDS_OUT);
```

Comment:
should we make this generic and not have two (trackedgroup, trackedaggregate) separate methods ?